src/plugin.js

```diff
@@ -1,69 +1,80 @@
 'use strict';
 var Joi = require('joi');
+var _ = require('lodash');
+var otSchemas = require('ot-schemas');
+var SCHEMAS = _.get(otSchemas, 'searchApi.helpers.request', {});
 
-exports.register = function(plugin, options, next){
+var camelCaseSchema = _.mapKeys(SCHEMAS.COMBINED_SCHEMA, _.rearg(_.camelCase, 1));
+var keyMapping = _.mapKeys(_.mapValues(camelCaseSchema, function(value, key) { return key; }), function(value) { return _.toLower(value);});
```

Comment:
basically, these four lines of code here that generate keyMapping should be done on the consumer (in your case, our internal freetext-api) and then passed in as a option to this plugin. Stuff like this shouldn't be in this plugin

Reply:
You could even do something like this in the freetext-api:
```
camelCase: keyMappings
```
where you generate the keyMappings in the api, then pass the generated config to the the `camelCase` option, and if options are present, use them.